### PR TITLE
[serve] Add Ray API stability annotations

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Add stability annotations recently generalized from RLlib.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@ericl this looks pretty good. I don't think we have anything that fits into the `DeveloperAPI` category (yet)

**Review by simon-mo — APPROVED** [review]
LGTM looks great

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -583,6 +583,7 @@ def get_handle(
         return handle
 
 
+@ray.util.annotations.PublicAPI
```
Comment (ericl) [inline python/ray/serve/api.py:586]: `from ray.util.annotations import PublicAPI`

`@PublicAPI`

**Review by ericl — CHANGES_REQUESTED** [review]
Please remove `ray.util.annotations.` prefix for consistency with other uses of the annotation

**Review by ericl — APPROVED** [review]

**Pull request merged**